src/main/kotlin/lotto/domain/WinningNumbers.kt

```diff
@@ -0,0 +1,22 @@
+package lotto.domain
+
+class WinningNumbers(winningNumbersText: String) {
+    private val winningNumbers: Set<LottoNumber>
+
+    init {
+        winningNumbers = winningNumbersText.split(TEXT_WINNING_NUMBERS_DELIMITER)
+            .map { LottoNumber.from(it.toInt()) }
+            .toSet()
+        require(winningNumbers.size == LottoNumbers.LOTTO_NUMBER_COUNT)
+    }
+
+    fun rank(lotto: Lotto): Rank {
+        return Rank.from(lotto.numbers.count {
+            winningNumbers.contains(it)
+        })
```

Comment:
```suggestion
        return Rank.from(lotto.compare(winningNumbers))
```

결국 winningNumbers도 Lotto로 취급 할 수 있어보여요. Lotto와 winningNumber에 대한 비교를 Lotto 안에서 한다면 로직이 간결해지고, Lotto 객체가 하는 일에 대해서 더욱 명확해지지 않을까요?